<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta content="Maciej Bak - About" name="description">
  <meta content="Maciej Bak" name="author">

  <title>Maciej Bąk - About</title>

  <link rel="stylesheet"
    href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css"
    integrity="sha384-9gVQ4dYFwwWSjIDZnLEWnxCjeSWFphJiwGPXr1jddIhOegiu1FwO5qRGvFXOdJZ4"
    crossorigin="anonymous">
  <link rel="stylesheet"
    href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" />
  <link rel="stylesheet" href="style.css" />
</head>
<body>

<div class="jumbotron jumbotron-fluid">
  <div class="container">
    <h1 class="mb-4">Maciej Bąk, PhD</h1>
    <p class="lead">
      Scientific Software Engineering Expert
    </p>
  </div>
</div>

<div class="container mb-5">
  <div class="row">
    <div class="col-sm-8 col-lg-3 mb-4">
      <img src="profile.jpeg" alt="Maciej Bąk - Profile picture"
        class="img-fluid" style="width:250px;" />
    </div>
    <div class="col-sm-4 col-lg-3">
      <ul class="fa-ul">
        <li>
          <a href="https://github.com/angrymaciek" target="_blank">
            <i class="fa-li fa fa-github"></i> GitHub
          </a>
        </li>
        <li>
          <a href="https://twitter.com/angrymaciek" target="_blank">
            <i class="fa-li fa fa-twitter"></i> Twitter
          </a>
        </li>
        <li>
          <a href="https://scholar.google.com/citations?user=i-aMGygAAAAJ&hl=en" target="_blank">
            <i class="fa-li fa fa-graduation-cap"></i> Google Scholar
          </a>
        </li>
        <li>
          <a href="https://orcid.org/0000-0003-1361-7301" target="_blank">
            <i class="fa-li fa fa-graduation-cap"></i> ORCID
          </a>
        </li>
        <li>
          <a href="https://www.researchgate.net/profile/Maciek_Bak" target="_blank">
            <i class="fa-li fa fa-graduation-cap"></i> ResearchGate
          </a>
        </li>
        <li>
          <a href="https://www.linkedin.com/in/bakmaciej" target="_blank">
            <i class="fa-li fa fa-linkedin"></i> LinkedIn
          </a>
        </li>
      </ul>
    </div>
    <div class="col-sm-12 col-lg-6">
      <p>
        I'm currently a freelance computational researcher.
        Previous affiliations include Biozentrum &
        Swiss Institute of Bioinformatics (Universitat Basel) as well as
        Department of Mathematics (City, University of London).
      </p>
      <p>
        I am interested in various computational approaches to study and combat cancer.
        Mostly passionate about the application of math/stat modelling
        and machine learning methods to solve biological problems.
        High-quality scientific software engineering is my speciality.
        In my (little) spare time I train combat systems, design and develop
        my very own horror game or read about mathematics and philosophy.
      </p>
      <p>
        If you'd like to collaborate with me please check out the <a href="core.html">CORE</a>.
      </p>
    </div>
  </div>
</div>

<div class="container mb-5">
  <div class="row">
    <div class="col-lg mb-5">
      <h4><i class="fa fa-book"></i> scientific articles</h4>
      <br>
      <ol>
        <li>
          Stein A., Kizhuttil R., Bak M., Noble R. (2024).
          <strong>Selective sweep probabilities in spatially expanding populations.</strong>
          <em>Nature Communications (under revision)</em>,
          <a href="https://doi.org/10.1101/2023.11.27.568915">DOI</a>
        </li>
        <br>
        <li>
          Szewczyk B., Japtok J., Held A., Dash P. B., Bak M., et al. (2024).
          <strong>Activation of polo-like kinase 1 expression in spinal motor neurons as shared molecular mechanisms across different types of familial ALS.</strong>
          <em>Acta Neuropathologica (under revision)</em>
        </li>
        <br>
        <li>
          Katsantoni M., Gypas F., Herrmann J C., Burri D., Bak M., et al. (2024).
          <strong>ZARP: A user-friendly and versatile RNA-seq analysis workflow.</strong>
          <em>F1000Research</em>,
          <a href="https://doi.org/10.12688/f1000research.149237.1">DOI</a>
        </li>
        <br>
        <li>
          Bak M., et al. (2024).
          <strong>Frequent co-regulation of splicing and polyadenylation by RNA-binding proteins inferred with MAPP.</strong>
          <em>Nature Communications</em>,
          <a href="https://doi.org/10.1038/s41467-024-48046-1">DOI</a>
        </li>
        <br>
        <li>
          Colyer B., Bak M., Basanta D., Noble R. (2024).
          <strong>A seven-step guide to spatial, agent-based modelling of tumour evolution.</strong>
          <em>Evolutionary Applications</em>,
          <a href="https://doi.org/10.1111/eva.13687">DOI</a>
        </li>
        <br>
        <li>
          Bak M. (2023).
          <strong>Hypercomplex: abstract & fast header-only C++ template library for lattice-based cryptosystems in high-dimensional algebras.</strong>
          <em>Journal of Open Source Software</em>,
          <a href="https://doi.org/10.21105/joss.05272">DOI</a>
        </li>
        <br>
        <li>
          Bak M., Colyer B., Manojlović V., Noble R. (2023).
          <strong>Warlock: an automated computational workflow for simulating spatially structured tumour evolution.</strong>
          <em>arXiv</em>,
          <a href="https://doi.org/10.48550/arXiv.2301.07808">DOI</a>
        </li>
        <br>
        <li>
          Bak M.<sup>*</sup>, Rozlach A. (2023).
          <strong>Modeling microbial communities with evolutionary game theory.</strong>
          <em>Journal of Brief Ideas</em>,
          <a href="https://beta.briefideas.org/ideas/d292ad28e6818d20aad2286e1c9b6aa2">URL</a>
        </li>
        <br>
        <li>
          Kwiecien K., Majewski P., Bak M., et al. (2023).
          <strong>Alternative splicing is not a key source of chemerin isoforms diversity.</strong>
          <em>Molecular Biology Reports</em>,
          <a href="https://doi.org/10.1007/s11033-022-08174-7">DOI</a>
        </li>
        <br>
        <li>
          Ghosh S., Ataman M., Bak M., et al. (2022).
          <strong>CFIm-mediated alternative polyadenylation remodels cellular signaling and miRNA biogenesis.</strong>
          <em>Nucleic Acid Research</em>,
          <a href="https://doi.org/10.1093/nar/gkac114">DOI</a>
        </li>
        <br>
        <li>
          Bak M.<sup>*</sup>, Rozlach A. (2020).
          <strong>Moran Pycess: a Python package to simulate Moran processes driven by game theory.</strong>
          <em>Journal of Open Source Software</em>,
          <a href="https://doi.org/10.21105/joss.02643">DOI</a>
        </li>
        <br>
        <li>
          Kwiecien K., Brzoza P., Bak M., et al. (2020).
          <strong>The methylation status of the chemerin promoter region located from - 252 to + 258 bp regulates constitutive but not acute-phase cytokine-inducible chemerin expression levels.</strong>
          <em>Scientific Reports</em>,
          <a href="https://doi.org/10.1038/s41598-020-70625-7">DOI</a>
        </li>
        <br>
        <li>
          Hasler D., Rajyalakshmi M., Bak M., et al. (2020).
          <strong>The Alazami Syndrome-associated protein LARP7 guides U6 small nuclear RNA modification and contributes to splicing robustness.</strong>
          <em>Molecular Cell</em>,
          <a href="https://doi.org/10.1016/j.molcel.2020.01.001">DOI</a>
        </li>
        <br>
        <li>
          Gałan W., Bąk M., Jakubowska M. (2019).
          <strong>Host Taxon Predictor - A Tool for Predicting Taxon of the Host of a Newly Discovered Virus.</strong>
          <em>Scientific Reports</em>,
          <a href="https://doi.org/10.1038/s41598-019-39847-2">DOI</a>
        </li>
        <br>
        <li>
          Bak M., Jazwa A., Kasper L. et al. (2015).
          <strong>Involvement of microRNAs in the inflammatory pathways of pulmonary sarcoidosis.</strong>
          <em>Journal of Physiology and Pharmacology</em>,
          <a href="https://www.ncbi.nlm.nih.gov/pubmed/26579569">URL</a>
        </li>
        <br>
        <li>
          Jazwa A., Kasper L., Bak M., et al. (2015).
          <strong>Differential inflammatory microRNA and cytokine expression in pulmonary sarcoidosis.</strong>
          <em>Archivum Immunologiae et Therapiae Experimentalis</em>,
          <a href="https://doi.org/10.1007/s00005-014-0315-9">DOI</a>
        </li>
      </ol>
    </div>
    <div class="col-lg mb-4">
      <h4><i class="fa fa-code"></i> software repositories</h4>
      <p style="border-bottom: solid 1px;">Science</p>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/warlock">warlock</a>
        </dt>
        <dd class="col-8">
          Snakemake workflow dedicated to deme-based oncology modeling.
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/MAPP">MAPP</a>
        </dt>
        <dd class="col-8">
          Computational method to identify binding motifs for RBPs that shape pre-mRNA processing.
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/zavolanlab/bindz-rbp">bindz</a>
        </dt>
        <dd class="col-8">
          Tool for predicting binding sites of RNA-binding proteins in transcripts.
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/zarp">ZARP</a>
        </dt>
        <dd class="col-8">
          Snakemake pipeline for automated RNA-seq data processing.
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/angry-moran-simulator">Moran Pycess</a>
        </dt>
        <dd class="col-8">
          Python package for simulations according to
          evolutionary game theory.
        </dd>
      </dl>
      <p style="border-bottom: solid 1px;">Others</p>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/hypercomplex">Hypercomplex</a>
        </dt>
        <dd class="col-8">
          Abstract & fast C++ template library for lattice-based cryptosystems in high-dimensional algebras.
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/efcomp">efcomp</a>
        </dt>
        <dd class="col-8">
          Collection of efficient statistical computing methods (interfacing from R/Python to Cpp).
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/angry-runner">Angry Runner</a>
        </dt>
        <dd class="col-8">
          Docker recipe for a self-hosted GitHub Actions runner (CI/CD).
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/angry-droid">Angry Droid</a>
        </dt>
        <dd class="col-8">
          Scientific computing platform setup for Android.
        </dd>
      </dl>
      <dl class="row mb-4">
        <dt class="col-3 mr-2">
          <a href="https://github.com/AngryMaciek/bubblefuck">Angry Bubblef*ck</a>
        </dt>
        <dd class="col-8">
          Bubble sort implemented in brainf*ck language.
        </dd>
      </dl>
    </div>
  </div>
</div>

<footer>
  <div class="container text-center">
    <p>
      <i class="fa fa-copyright"></i> 2025 Maciej Bąk
    </p>
    <p>
      Hosted by GitHub, Licensed by MIT
    </p>
  </div>
</footer>
</body>
</html>
